### PR TITLE
azure - sql vulnerability scan filter

### PR DESCRIPTION
**Pull request by stefangordon**
```
policies:
  - name: sql-vulnerability-assessment
    description: |
      find sql servers that do not
      have recurring vulnerability scans
      enabled
    resource: azure.sql-server
    filters:
      - type: vulnerability-assessment
        enabled: false
```

Related to #6551 

Also add filter to administrators test to fix functional test failure.

### CODE REVIEW

File: tools/c7n_azure/c7n_azure/resources/sqlserver.py [hunk 1]
```diff
@@ -136,6 +137,77 @@ def __call__(self, i):
...
+
+    schema = type_schema(
+        'vulnerability-assessment',
+        required=['type', 'enabled'],
+        **{
+            'enabled': {"type": "boolean"},
+        }
+    )
+
+    log = logging.getLogger('custodian.azure.sqldatabase.VulnerabilityAssessmentFilter')
```
Comment (logachev) [inline tools/c7n_azure/c7n_azure/resources/sqlserver.py:169]: I know we have a mix of `camelCase` and `dash-case`.. Should we pick a standard one?

File: tools/c7n_azure/c7n_azure/resources/sqlserver.py [hunk 2]
```diff
@@ -136,6 +137,77 @@ def __call__(self, i):
...
+
+    def __init__(self, data, manager=None):
+        super(VulnerabilityAssessmentFilter, self).__init__(data, manager)
+        self.enabled = self.data.get('enabled')
```
Comment (logachev) [inline tools/c7n_azure/c7n_azure/resources/sqlserver.py:173]: I think we used `['enabled']` for required params

File: tools/c7n_azure/c7n_azure/resources/sqlserver.py [hunk 3]
```diff
@@ -136,6 +137,77 @@ def __call__(self, i):
...
+        )
+        if exceptions:
+            raise exceptions[0]
+        return resources
+
+    def _process_resource_set(self, resources, event=None):
+        client = self.manager.get_client()
+        result = []
+        for resource in resources:
+            if 'c7n:vulnerability_assessment' not in resource['properties']:
```
Comment (logachev) [inline tools/c7n_azure/c7n_azure/resources/sqlserver.py:191]: Is it not available via resource graph?

File: tools/c7n_azure/c7n_azure/resources/sqlserver.py [hunk 4]
```diff
@@ -136,6 +137,77 @@ def __call__(self, i):
...
+                # there can only be a single instance named "Default".
+                if va:
+                    resource['c7n:vulnerability_assessment'] = \
+                        va[0].serialize(True).get('properties', {})
+                else:
+                    resource['c7n:vulnerability_assessment'] = {}
+
+            if resource['c7n:vulnerability_assessment']\
+                    .get('recurringScans', {})\
+                    .get('isEnabled') == self.enabled:
```
Comment (logachev) [inline tools/c7n_azure/c7n_azure/resources/sqlserver.py:205]: `.get('isEnabled', False)` ?

**Review by logachev — APPROVED** [review]

File: tools/c7n_azure/c7n_azure/resources/sqlserver.py (hunk 3 above)
Comment (stefangordon) [reply tools/c7n_azure/c7n_azure/resources/sqlserver.py:191]: I couldn't get it there and the docs didn't mention it like the others so I prefixed it ... not an exact science I guess.

File: tools/c7n_azure/c7n_azure/resources/sqlserver.py (hunk 4 above)
Comment (stefangordon) [reply tools/c7n_azure/c7n_azure/resources/sqlserver.py:205]: I was thinking if it comes back null you'd want it to trigger either way, but your right null can just be assumed the same as false I guess.

File: tools/c7n_azure/c7n_azure/resources/sqlserver.py (hunk 1 above)
Comment (kapilt) [reply tools/c7n_azure/c7n_azure/resources/sqlserver.py:169]: its generally lower case for loggers re standard

**Pull request merged**